packages/flutter_markdown/lib/src/builder.dart

```diff
@@ -131,9 +160,12 @@ class MarkdownBuilder implements md.NodeVisitor {
   /// The base directory holding images referenced by Img tags with local or network file paths.
   final String? imageDirectory;
 
-  /// Call when build an image widget.
+  /// Call when building an image widget when no sizing information is available.
   final MarkdownImageBuilder? imageBuilder;
 
+  /// Call when building an image widget with sizing information available.
```

Comment:
Same here. It's called if passed; nothing in the code is conditional on the availability of any information.

Reply:
Also, both of these need to document that both can't be passed. Anything that is asserted as a calling requirement in code needs to be clearly and explicitly communicated in the API docs.